pkg/controller/render/render_controller.go

```diff
@@ -431,12 +432,14 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 	if err != nil {
 		return err
 	}
+	sort.SliceStable(mcs, func(i, j int) bool { return mcs[i].Name < mcs[j].Name })
```

Comment:
what is this for?

Reply:
I feel like I remember this existing but it claims you're adding this, do we not do this somewhere else?

Reply:
unit tests were failing with `error finding pools for machineconfig` and `Action update machineconfigpools has wrong object`

the order of elements in the slice was being altered in `syncMachineConfigPool` from the lister

Reply:
> I feel like I remember this existing but it claims you're adding this, do we not do this somewhere else?

We do it later in like MergeMachineConfigs, but there is some weird non-determinism that we hit a lot in this PR during tests for some reason. see: https://github.com/openshift/machine-config-operator/pull/4060#issuecomment-1855012970 

I have been seeing it intermittently, e.g. https://github.com/openshift/machine-config-operator/pull/4003#issuecomment-1822032846 so I know this PR didn't introduce it, it just seems like it happens way more often here. It might be intended behavior ( like how they intentionally don't want you to depend on go map order so they shuffle it) but I haven't done the due diligence, I'm a slob. :smile: 

I wasn't being prescriptive on how we accommodate this, I just wanted to prove why it was breaking -- it's possible we could mangle the test fixture to just test for presence not order but that also might be a weird one-off.